File: Unity Project/Assets/Scripts/Data/DataController.cs
namespace CardGame.Data
{
    public class DataController
    {
        private bool m_simulateNativeServer;
        private IDataProvider m_dataProvider;

        public DataController(bool isLocalMode, bool simulateNativeServer)
        {
            m_simulateNativeServer = simulateNativeServer;
            m_dataProvider = m_dataProvider = isLocalMode ? new LocalDataProvider() : new RemoteDataProvider(m_simulateNativeServer);    
        }
        
        public PlayerData GetPlayer(int deckSize, bool isEnemy) => m_dataProvider.GetPlayer(deckSize, isEnemy);

        public void IncreaseCardPopularity(int dataBaseID) => m_dataProvider.IncreaseCardPopularity(dataBaseID);
    }
}

File: SDK Project/app/src/main/java/com/cardgame/sdk/data/DataController.java
package com.cardgame.sdk.data;

import android.util.Log;
import com.google.gson.Gson;

public class DataController {

    private static final int CARD_SIZE = 7;

    private static DataRetriever dataRetriever;
    public static String URL;

    public void Init(boolean simulateServer, String url) {
        dataRetriever = new DataRetriever(simulateServer);
        UpdateUrl(url);
    }

    public void UpdateUrl(String url) {
        URL = url;
    }

    public String GetPlayerData(boolean isEnemy)
    {
        if (dataRetriever == null)
        {
            Log.e("[SDK]", "DataRetriever is not initialized");
            return null;
        }

        PlayerData data = dataRetriever.GetPlayer(CARD_SIZE, isEnemy);
        Gson gson = new Gson();

        return gson.toJson(data);
    }

    public void IncreaseCardPopularity(int dataBaseID)
    {
        if (dataRetriever == null)
        {
            Log.e("[SDK]", "DataRetriever is not initialized");
            return;
        }

        dataRetriever.IncreaseCardPopularity(dataBaseID);
    }
}
